src/CodeGen_Hexagon.cpp

```diff
@@ -1346,7 +1346,138 @@ void CodeGen_Hexagon::visit(const Call *op) {
             return;
         }
     }
-    CodeGen_Posix::visit(op);
+
+    if (op->is_intrinsic(Call::predicated_store)) {
+        internal_assert(op->args.size() == 3) << "predicated_store takes three arguments: {store addr, predicate, value}\n";
```

Comment:
Perhaps that should be implemented in CodeGen_LLVM, with those functions virtual?